### PR TITLE
Update resource_name/provides cops to detect new Chef 16.2+ behavior

### PR DESCRIPTION
**Pull request by lamont-granquist**

### CODE REVIEW

**Comment by lamont-granquist** [comment]
Yes, but that commit was for "allow Chef-16 usage".  Which means to not throw an offense on it and to not force people towards using only the Chef-15 backcompat usage.

File: spec/rubocop/cop/chef/deprecation/hwrp_without_resource_name_and_provides_spec.rb [hunk 1]
```diff
@@ -73,27 +73,9 @@ class UlimitRule < Chef::Resource
...
   end
 
-  # This is correct usage for >= Chef-16, but this will fail on <= Chef-15 (this should become the autocorrect case in >= Chef-16)
-  # This is allowed here under the assumption that the person is writing cookbooks for >= Chef-16
-  it 'registers an offense when a Resource derived HWRP does not define a resource_name' do
-    expect_no_offenses(<<~RUBY)
-    class Chef
-      class Resource
-        class UlimitRule < Chef::Resource
-          provides :ulimit_rule
```
Comment (lamont-granquist) [inline spec/rubocop/cop/chef/deprecation/hwrp_without_resource_name_and_provides_spec.rb]: yeah you probably shouldn't delete this.  we want to allow this.

File: spec/rubocop/cop/chef/deprecation/resource_uses_only_resource_name_spec.rb [hunk 2]
```diff
@@ -96,25 +96,13 @@
...
   it "doesn't register an offense when a resource has resource_name and provides" do
     expect_no_offenses(<<~RUBY, '/mydevdir/cookbooks/my_cookbook/resources/foo.rb')
       resource_name :my_cookbook_foo
       provides :my_cookbook_foo
     RUBY
   end
 
-  it "doesn't register an offense when a resource has just provides (and the cookbook name doesn't match)" do
-    expect_no_offenses(<<~RUBY, '/mydevdir/cookbooks/my_cookbook/resources/foo.rb')
-      provides :my_cookbook_bar
```
Comment (lamont-granquist) [inline spec/rubocop/cop/chef/deprecation/resource_uses_only_resource_name_spec.rb]: same with these two, this should be legal syntax

File: spec/rubocop/cop/chef/deprecation/hwrp_without_resource_name_and_provides_spec.rb (hunk 1 above)
Comment (tas50) [reply spec/rubocop/cop/chef/deprecation/hwrp_without_resource_name_and_provides_spec.rb]: At the moment we're enforcing a Chef 15 compatible syntax which requires that be there.

**Pull request merged**